nbclient/client.py

```diff
@@ -358,12 +433,11 @@ def start_kernel_manager(self):
             self.start_kernel_manager()
 
         if not self.km.has_kernel:
-            await self.start_new_kernel_client(**kwargs)
+            await self.async_start_new_kernel_client(**kwargs)
         try:
             await yield_(None)  # would just yield in python >3.5
         finally:
-            self.kc.stop_channels()
-            self.kc = None
+            self._cleanup_kernel()
```

Comment:
Here we might use an async version of `_cleanup_kernel`, since we are in an async function.